app/controllers/drivers_controller.rb

```diff
@@ -5,7 +5,24 @@ def index
     @model_attributes = model_attributes(Driver)
   end
 
+  def new
+    @driver = Driver.new
+  end
+
+  def create
+    driver = Driver.new(driver_params)
+
+    if driver.save
+      redirect_to drivers_path
+    else
+      render 'new'
```

Comment:
In this case `@driver` will be nil, and you can get an exception during the rendering of the view template

Reply:
good catch!

Reply:
@benrobinson-ms @alexandersysuiev-ms 
im trying to understand this, so in the case that the created driver is invalid, the `@driver` will be nil?

Reply:
As you didn't define the `@driver` variable in the `create` action it will be nil in the view
Here
```
= form_with model: @driver do |f|
```

Reply:
Why did you resolve this comment? Did you set an instance variable?

Reply:
@mmmmaximus this one isnt actually resolved yet since you never set the instance variable for this to work.
Id recommend not being too gungho with resolving comments since you didnt get it totally correct.
you need to set `@driver` inside your create too otherwise an error will still render a blank new i think

Reply:
> you need to set `@driver` inside your create too otherwise an error will still render a blank new i think

Would it still be needed since I have already added the `helper_method :driver`?

Reply:
have you checked this behaviour?
on error do you get render `new` with all the same input data or does it go back to an empty form?

Reply:
> on error do you get render `new` with all the same input data or does it go back to an empty form?

just understood, when `new` is rendered with an invalid driver, the same input data remains